src/pybind/mgr/rook/module.py

```diff
@@ -271,23 +268,21 @@ def describe_service(self, service_type=None, service_id=None, node_name=None, r
             sd.status = status
             sd.status_desc = p['phase']
 
-            if sd.service_type == "osd":
-                sd.service_instance = "%s" % p['labels']["ceph-osd-id"]
-            elif sd.service_type == "mds":
-                sd.service = p['labels']['rook_file_system']
-                pfx = "{0}-".format(sd.service)
-                sd.service_instance = p['labels']['ceph_daemon_id'].replace(pfx, '', 1)
-            elif sd.service_type == "mon":
-                sd.service_instance = p['labels']["mon"]
-            elif sd.service_type == "mgr":
-                sd.service_instance = p['labels']["mgr"]
-            elif sd.service_type == "nfs":
-                sd.service = p['labels']['ceph_nfs']
-                sd.service_instance = p['labels']['instance']
-                sd.rados_config_location = self.rook_cluster.get_nfs_conf_url(sd.service, sd.service_instance)
-            elif sd.service_type == "rgw":
-                sd.service = p['labels']['rgw']
-                sd.service_instance = p['labels']['ceph_daemon_id']
+            if sd.daemon_type == "osd":
+                sd.daemon_id = "%s" % p['labels']["ceph-osd-id"]
+            elif sd.daemon_type == "mds":
+                pfx = "{0}-".format(p['labels']['rook_file_system'])
+                sd.daemon_id = p['labels']['ceph_daemon_id'].replace(pfx, '', 1)
+            elif sd.daemon_type == "mon":
+                sd.daemon_id = p['labels']["mon"]
+            elif sd.daemon_type == "mgr":
+                sd.daemon_id = p['labels']["mgr"]
+            elif sd.daemon_type == "nfs":
+                sd.daemon_id = p['labels']['instance']
+                sd.rados_config_location = self.rook_cluster.get_nfs_conf_url(
```

Comment:
`rados_config_location` is not present in  `DaemonDescription` class (it is only present in `ServiceDescription` class)

Reply:
The `container_image_id` and `container_image_name` are not set in the `DaemonDescription` object. why?

Reply:
Is a pity that in Rook the "daemon_id"  is a different thing for each type pods. Maybe it would be worth to unify this in Rook and to have a "daemon_id" in each kind of relevant pod to avoid the external build of this attribute.